src/main/scala/mesosphere/marathon/core/leadership/LeaderShipModule.scala

```diff
@@ -0,0 +1,33 @@
+package mesosphere.marathon.core.leadership
+
+import akka.actor.{ ActorRef, ActorRefFactory, Props }
+import mesosphere.marathon.core.leadership.impl.{
+  ActorLeadershipCoordinator,
+  LeadershipCoordinationActor,
+  WhenLeaderActor
+}
+
+class LeadershipModule(actorRefFactory: ActorRefFactory) {
+
+  private[this] var whenLeaderRefs = Set.empty[ActorRef]
+  private[this] var started: Boolean = false
+
+  def startWhenLeader(props: => Props, name: String, considerPreparedOnStart: Boolean = true): ActorRef = {
```

Comment:
Damn. Definitely. The documentation was in the trait of this module and I got rid of the traits. 

Just to be clear: Are we talking about coding overhead or performance? Performance would be fine but I didn't want to make the common case (we do not need special initialization) more complicated then necessary. Also every top level actor would then have to know about PreparationMessages. 

It would probably add the following coding overhead:
* one line for very actor without special initialization logic in production code
* two lines (send PrepareToStart, expect Prepared) in many tests using actors which would normally not require any initialization logic. (Maybe unnecessary if its a NOOP but strange not to test that.)